flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/BufferConsumer.java

```diff
@@ -44,6 +44,8 @@
 
 	private int currentReaderPosition;
 
+	private final boolean isExactlyOnceBarrier;
```

Comment:
It seems a bit strange to see this field in `BufferConsumer`. 
After discussing offline, we introduce another enum type for identifying the specific `AbstractEvent` instance inside `Buffer` without deserialization. Then the required components can judge the type from buffer to make some improvements. At the beginning we only focus on `CheckpointBarrier` event and make others unknown.